packages/patternfly-3/patternfly-react/src/components/DateTimePicker/TimePicker.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl, InputGroup, Icon, OverlayTrigger, Popover } from '../../';
+import TimeInput from './TimeComponents/TimeInput';
+
+class TimePicker extends React.Component {
+  getDateFromTime = time => {
+    if (Date.parse(time)) {
+      return new Date(time);
+    }
+    return new Date(`1/1/1 ${time}`);
+  };
+
+  formatDate = time => {
+    const { locale } = this.props;
+    const options = { hour: '2-digit', minute: '2-digit' };
+    if (Date.parse(time)) {
+      const parsedValue = new Date(time);
+      return parsedValue.toLocaleString(locale, options);
+    } else if (Date.parse(`1/1/1 ${time}`)) {
+      const parsedValue = new Date(`1/1/1 ${time}`);
+      return parsedValue.toLocaleString(locale, options);
+    }
+    return time;
+  };
+  state = {
```

Comment:
can you move it to the top of the file?

Reply:
moved the state, and moved the functions into a helper file